app/controllers/peer_reviews_controller.rb

```diff
@@ -157,7 +157,21 @@ def assign_groups
         return
       end
     when 'unassign'
-      PeerReview.unassign(selected_reviewee_group_ids, reviewers_to_remove_from_reviewees_map)
+      reviews_deleted, deleted_count, undeleted_reviews = PeerReview.unassign(selected_reviewee_group_ids,
+                                                                              reviewers_to_remove_from_reviewees_map)
+      if !reviews_deleted && deleted_count == 0
+        flash_now(:error, t('peer_reviews.errors.cannot_unassign_any_reviewers'))
+        return
+      elsif !reviews_deleted && undeleted_reviews.length <= 5
+        flash_now(:error, t('peer_reviews.errors.cannot_unassign_all_reviewers',
+                            deleted_count: deleted_count.to_s, undeleted_reviews: undeleted_reviews.join(', ')))
+      elsif !reviews_deleted && undeleted_reviews.length > 5  # Truncate the list of undeleted reviews to 5
```

Comment:
This case is okay, but again can be merged into the previous one as follows:

1. Always call `undeleted_reviews.first(5)` (it will keep all elements if there are fewer than 5)
2. Reuse the same error message (`cannot_unassign_all_reviewers_many`), but then when there are > 5, append a second message that just says "Additional <count> not shown". Since this is a more generic message (and one that I'd hope to start using elsewhere in the application), you can add that message to `config/locales/common/en.yml`

Reply:
I'm a little bit confused with what I should do here. In the case that `undeleted_reviews.length > 5`, should I just print `peer_reviews.errors.cannot_unassign_all_reviewers`, but then append an extra "Additional not shown" at the end? Also, would this imply that I should delete `peer_reviews.errors.cannot_unassign_all_reviewers_many`, since it does not serve any purpose anymore?

Thanks for the feedback 😄

Reply:
Yes---I meant use the error message `peer_reviews.errors.cannot_unassign_all_reviewers`, with the appended text! And yes, you should *delete* the `peer_reviews.errors.cannot_unassign_all_reviewers_many` message